lib/broccoli/ember-app.js

```diff
@@ -1845,6 +1845,30 @@ function addOutputFile(container, assetPath, options) {
     container[outputFile] = [];
   }
 
+  if (~container[outputFile].indexOf(assetPath)) {
+    var message = [
+      'You have specified a duplicate asset via `app.import(' + assetPath + ')`.',
+      'This can result in very strange behavior in your application as the',
+      'asset historically was included twice. Multiple versions of the same asset',
+      'are also clobbered by the last version added to the Broccoli tree, meaning',
+      'that it only ever included the version of the asset which was last added',
+      'to the Broccoli tree. This should be treated as a bug to address. This',
+      'typically occurs when an asset is imported by a nested addon. Ember CLI',
+      'no longer includes multiple copies by default and instead is using only',
+      'the _last_ occurrence imported.'
+    ];
+    this.project.ui.writeWarnLine(message.join(' '));
```

Comment:
> We should basically just do the right thing.

The "right thing" here is nebulous. If we're completely silent then the user goes about their merry way and is woefully unaware that we've enforced the Highlander Rule on them with `broccoli-merge-trees` and `{ overwrite: true }` (well before `app.import` gets run). Then, to prevent further trolling, we make it so that we at least don't include the asset multiple times in the bundled output. Basically we let you shoot yourself in the foot then we hand you a bandaid.

I'm modifying to `logger.info` but at some point we need to start making plans to move the ecosystem.
 
> we should choose a dedupe strategy 

Dedupe strategy is already defined and tested: last import wins (by array order, not invocation order because `prepend`).